processing/src/main/java/io/druid/query/search/search/AutoStrategy.java

```diff
@@ -0,0 +1,56 @@
+package io.druid.query.search.search;
+
+import com.metamx.emitter.EmittingLogger;
+import io.druid.collections.bitmap.ImmutableBitmap;
+import io.druid.query.dimension.DimensionSpec;
+import io.druid.segment.QueryableIndex;
+import io.druid.segment.Segment;
+
+import java.util.List;
+
+public class AutoStrategy extends SearchStrategy
+{
+  public static final String NAME = "auto";
+
+  private static final EmittingLogger log = new EmittingLogger(AutoStrategy.class);
+
+  public AutoStrategy(SearchQuery query)
+  {
+    super(query);
+  }
+
+  @Override
+  public List<SearchQueryExecutor> getExecutionPlan(SearchQuery query, Segment segment)
+  {
+    final QueryableIndex index = segment.asQueryableIndex();
+
+    if (index != null) {
+      final ImmutableBitmap timeFilteredBitmap = IndexOnlyStrategy.makeTimeFilteredBitmap(index,
+                                                                                          segment,
+                                                                                          filter,
+                                                                                          interval);
+      final Iterable<DimensionSpec> dimsToSearch = getDimsToSearch(index.getAvailableDimensions(),
+                                                                   query.getDimensions());
+
+      // Choose a search query execution strategy depending on the query.
+      // Index-only strategy is selected when
+      // 1) there is no filter,
+      // 2) the total cardinality is very low, or
+      // 3) the filter has a very low selectivity.
```

Comment:
It is equivalent now, but only by luck. We are planning to add some filters in the future that don't ever support bitmaps (like filters that filter based on expressions). Those would return false from "supportsBitmapIndex" even if the underlying dimensions did have indexes, since they'd always want to operate on a row by row basis.